### DISCUSSION TITLE
Paid Elastic License

### DISCUSSION THREAD

**Discussion by fluffyunicorn24** [Security-Onion-Solutions/securityonion#5051] [Unsupported Versions]
Good evening everyone,

My job may potentially purchase Elastic’s Endgame EDR solution. This license will also come with Elastic Security.  Has anyone configured this before? We were able to add our trial license, but data will not populate in the Elastic Security Dashboard. We are also unable to do any of the machine learning. 

I know there are other solutions that we can use outside of Security Onion, but they have asked me to specifically see if this was possible. 

I really appreciate any of the help you can provide.

**Comment by wdpless** [comment]
Were you able to get the detections page to load?

**Reply by fluffyunicorn24** [reply]
Yes, but I don’t think the data is being read. In the Network section I am seeing traffic in Flow but not DNS or HTTP.

**Reply by Jackson-Pollock** [reply]
I tried Elastic Security in Security Onion but there are two roadblock to me due to which I can't proceed: 
1- SO is non-compliant with Elastic common schema (ECS).
2- EQL doesn't support cross cluster search (CCS) however this is solved in 7.14 release.

**Reply by fluffyunicorn24** [reply]
Ok, so you made me feel better that I am not completely off on this.  I have found both of those to be the primary issues too.  I did an auto-follow CCR and was able to get flow data into Flow, but then I am still running into the mapping issue.  I am might try to put filebeats in the home directory and see what that does.

**Reply by TOoSmOotH** [reply]
Right now we can't follow ECS on some inputs because the parsing is not there yet. We are working to push some upstream changes but not sure when those will take place. The issue with the current official ECS is the data is compartmentalized per input vs what SO tries to do which is make it standard across everything. The only way to really address this is with a TON of aliases which we would need to add. Between the upstream changes and the aliases our hope is to be full ECS compliant at some point. Our original plan was to switch to full ECS in 2.3.60 but a ton of queries would have needed to be changed and the flow when hunting is just not right. 

An example of this is designating an alert as an alert. If you use ECS for Suricata it comes in as event.kind=alert. Zeek notice and weird also come in as event.kind=alert. So to find all "alerts" in your environment you would search for event.kind=alert. Although notice and weird can be useful at times, I would not call those alerts so to speak. If you use Suricata for meta data generation there is a ton missing from the current ECS implementation. 

So right now Elastic Security is not supported. Endgame data can be shipped to logstash on the manager although I have not tested the module.

**Reply by fluffyunicorn24** [reply]
Thank you so much!  I completely understand that this would be much harder to switch over since SO has so many components to it.  Hopefully Elastic comes up with a simpler way to manually identify the information it is looking for.  

I really appreciate everything SO does.  I never realized how difficult it was to configure an NSM until I had to do it manually.  All the backend work that is done is impressive.

**Comment by wdpless** [comment]
to get security detections to access the right indices have you tried:
In Kibana -> Stack Management -> Kibana(Advanced Settings) -> Security Solution(Elasticsearch indices) then add "`,*so-*`" ?  I was able to do this on another stack but am not yet to this step on my security onion setup yet.
![Screenshot from 2021-08-05 13-24-40](https://user-images.githubusercontent.com/8844528/128394183-5bdfd51a-37b3-4347-a48c-2d9b53be7217.png)


I am strugling getting the detection page to load. Did you have to build a Kibana cert on Elasticsearch container using the cert-util tool then carry that cert and the elasticsearch root ca.pem over to the kibana.yml file? My thought is that elastic requires the security app in kibana to be encrypted so that the api can query securely. Below is the screen I'm at.
Or am I going in the wrong direction?
![Screenshot from 2021-08-05 13-26-37](https://user-images.githubusercontent.com/8844528/128394423-fb8dfdbd-f6ad-4794-a6ae-0a376cc53711.png)

**Reply by fluffyunicorn24** [reply]
Ok so here is what I have done so far:

You will need to add the ":so-* index to the Elastic Security indices (Kibana → Stack Management → Advanced Settings → securitySolution:defaultIndex) 

You will need to set up CCR, but this may have broke my system.  I tried a couple of ways and one of them worked and one of them broke so I am retesting this now.

You will also need to set up a user account that has all the permissions.  I created an detection role (Kibana → Stack Management → Security →  Roles) and assign "manage" to cluster privileges, the appropriate indices with privileges as "all", and in Kibana put your space with privileges as "all"

You will also need to put the xpack.encryptedsavedobjects.encryptionkey in your kibana.yml file and make sure in elasticsearch.yml xpack.security.enabled is set to true 

With all of this I was able to get data in flow but not other sections.  I am still working it out.

**Reply by wdpless** [reply]
This setting got me to the page "xpack.encryptedsavedobjects.encryptionkey". Thank you!

**Reply by fluffyunicorn24** [reply]
No problem. Just append that to your Kibana.yml and you should be golden.

**Reply by n0th1nt0lOOz** [reply]
You haven't installed the detection engine yet. After you install the engine, the detection page will be loaded
https://github.com/elastic/detection-rules

**Reply by rwaight** [reply]
@fluffyunicorn24 @wdpless, have either of you been able to get the `*:so-*` or `so-*` index patterns to persist in the `securitySolution:defaultIndex` configuration in Kibana?  Every time Kibana restarts, my advanced settings are set back to the SO default, so making the change is not persistent.

**Reply by weslambert** [reply]
Right now, default configuration is reapplied to ensure consistency, however, we do realize folks would like to make certain changes, and are working to make that more customizable.  One way to keep the settings in the time being would be to make the change and export the saved objects with `so-kibana-config-export`, then place the `saved_objects.ndjson` in `/opt/so/saltstack/local/salt/kibana/files`, and it should take precedence over the default.

**Reply by rwaight** [reply]
That worked, @weslambert.  I noticed that I had to apply the changes needed then quickly issue the `so-kibana-config-export` command, thank you very much!

**Comment by wdpless** [comment]
I rebuilt the securityonion cluster as true elastic cluster and not cross-cluster and detections seems to be working. The preview option when creating detection doesnt find logs unless the field is ecs but the detections are firing and alerts from non-ecs fields are showing up. 

The next step im going to try is to enable zeek and suricata filebeat modules to see if all network data arrives on the security - network page.
